### PR TITLE
Add starts_with function and operator

### PR DESCRIPTION
**Pull request by papparapa**
[#5311](https://github.com/duckdb/duckdb/issues/5311#issue-1446440132)
I removed iterations over the haystack's offsets from the existing `contains` function which performs integer comparison instead of using the slower `memcmp` function.

### CODE REVIEW

**Comment by Mytherin** [comment]
Thanks for the PR! I have cancelled the CI runs for now as the release is being processed so this does not take up CI time.

File: src/function/scalar/string/starts_with.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+static bool StartsWith(const string_t &haystack_s, const string_t &needle_s) {
+	auto haystack = (const unsigned char *)haystack_s.GetDataUnsafe();
+	auto haystack_size = haystack_s.GetSize();
+	auto needle = (const unsigned char *)needle_s.GetDataUnsafe();
+	auto needle_size = needle_s.GetSize();
+	if (needle_size == 0) {
+		// empty needle: always true
+		return true;
+	}
+	return StartsWith(haystack, haystack_size, needle, needle_size);
```
Comment (Mytherin) [inline src/function/scalar/string/starts_with.cpp:98]: It seems that this code inherits a lot from the `strstr` code, but `strstr` is inherently a much harder function to implement efficiently. Can this code not simply be a call to `memcmp` together with some length checks? That will likely be faster as well.

**Review by Mytherin — COMMENTED** [review]
Thanks for PR! Looks good. One comment:

File: src/function/scalar/string/starts_with.cpp (hunk 1 above)
Comment (papparapa) [reply src/function/scalar/string/starts_with.cpp:98]: Thank you for your advice Mytherin!
I'm a begginer about calculation performance.
I will use `memcmp` and make the code simpler.

**Comment by Mytherin** [comment]
Thanks!

**Comment by Alex-Monahan** [comment]
Thanks! Could you add it into our function docs @papparapa ? They are located here:
https://github.com/duckdb/duckdb-web/blob/master/docs/sql/functions/char.md

**Pull request merged**